src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1433,32 +1433,160 @@ Error executeRcppEngineChunk(const std::string& docId,
 Error executeStanEngineChunk(const std::string& docId,
                              const std::string& chunkId,
                              const std::string& code,
+                             const std::map<std::string, std::string>& chunkOptions,
                              json::JsonRpcResponse* pResponse)
 {
-   // TODO:
-   return Success();
+   // forward declare error
+   Error error = Success();
+   
+   // always ensure we emit a 'execution complete' event on exit
+   ChunkExecCompletedScope execScope(docId, chunkId);
+   
+   // prepare chunk directory
+   FilePath outputPath = notebook::chunkOutputPath(
+            docId,
+            chunkId,
+            notebook::ContextExact);
+
+   error = outputPath.removeIfExists();
+   if (error)
+      LOG_ERROR(error);
+
+   error = outputPath.ensureDirectory();
```

Comment:
I'm wondering why we fail forward throughout the implementation (some of these errors such as being unable to create the outputPath seem like they would be fatal). This function returns an error, if we just directly returned these errors would they be reported in some sane/appropriate fashion?